pkg/api/v1/types.go

```diff
@@ -116,6 +119,7 @@ type VMList struct {
 
 // VMSpec is a description of a VM
 type VMSpec struct {
+	// Domain is the actual libvirt domain in JSON format. When creating a VM, it is converted to XML.
```

Comment:
Removing the v1.DomainSpec and merge it with v1.VMSpec would also be an option for me. I see this more of a relict, when v1.DomainSpec and api.DomainSpec really were the same struct.